packages/table/src/TableHead.tsx

```diff
@@ -0,0 +1,63 @@
+import React from 'react';
+import HeaderRow from './HeaderRow';
+import TableHeader from './TableHeader';
+import { TableProps } from './Table';
+import { isComponentType } from '@leafygreen-ui/lib';
+import { useTableContext } from './TableContext';
+
+type TableHeaderProps<Shape> = Pick<TableProps<Shape>, 'columns'>;
+
+function TableHead<Shape>({ columns = [] }: TableHeaderProps<Shape>) {
+  const usingHeaderRow = React.useRef(false);
```

Comment:
What if we just require the use of `HeaderRow`? How much does allowing this be implicit get us? They still need to use a Fragment, so might as well just use a proper component?

@DesignerDave ^

Reply:
What if we moved `<HeaderRow />` into the Table body? I may be scope creeping here.

Reply:
Because then they can't just do an array of strings which was an option as well

Reply:
I don't feel too strongly here, though I think allowing an array of strings is a nice feature that I believe more implementations than not would be able to use.